pkg/operation/botanist/component/kubeapiserverexposure/kube_apiserver_ingress.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubeapiserverexposure
+
+import (
+	"context"
+
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/gardener/pkg/controllerutils"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	kubeapiserverconstants "github.com/gardener/gardener/pkg/operation/botanist/component/kubeapiserver/constants"
+	"github.com/gardener/gardener/pkg/utils"
+)
+
+const (
+	nginxIngressSSLPassthrough = "nginx.ingress.kubernetes.io/ssl-passthrough"
+)
+
+// IngressValues configure the kube-apiserver ingress.
+type IngressValues struct {
+	// Host is the host where the kube-apiserver should be exposed.
+	Host string
+	// IngressClassName is the name of the ingress class.
+	IngressClassName *string
+	// ServiceName is the name of the service the ingress is using
+	ServiceName string
+}
+
+// NewIngress creates a new instance of Deployer for the ingress used to expose the kube-apiserver.
+func NewIngress(c client.Client, objectKey client.ObjectKey, values IngressValues) component.Deployer {
```

Comment:
Why do I need to pass the object key/why is this configurable?

Reply:
Only the namespace is required, the name can be a constant.